src/number_theory/padics/padic_norm.lean

```diff
@@ -286,5 +285,86 @@ begin
     { exact_mod_cast hp.1.one_lt } }
 end
 
-end padic_norm
+/-- The `p`-adic norm of an integer `m` is one iff `p` doesn't divide `m`. -/
+lemma int_eq_one_iff (m : ℤ) : padic_norm p m = 1 ↔ ¬ (p : ℤ) ∣ m :=
+begin
+  nth_rewrite 1 ← pow_one p,
+  simp only [dvd_iff_norm_le, int.cast_coe_nat, nat.cast_one, zpow_neg, zpow_one, not_le],
+  split,
+  { intro h,
+    rw [h, inv_lt_one_iff_of_pos];
+    norm_cast,
+    { exact nat.prime.one_lt (fact.out _), },
+    { exact nat.prime.pos (fact.out _), }, },
+  { simp only [padic_norm],
+    split_ifs,
+    { rw [inv_lt_zero, ← nat.cast_zero, nat.cast_lt],
+      intro h, exact (nat.not_lt_zero p h).elim, },
+    { have : 1 < (p : ℚ) := by norm_cast; exact (nat.prime.one_lt (fact.out _ : nat.prime p)),
+      rw [← zpow_neg_one, zpow_lt_iff_lt this],
+      have : 0 ≤ padic_val_rat p m, simp only [of_int, nat.cast_nonneg],
+      intro h,
+      rw [← zpow_zero (p : ℚ), zpow_inj];
+      linarith, } },
+end
+
+lemma int_lt_one_iff (m : ℤ) : padic_norm p m < 1 ↔ (p : ℤ) ∣ m :=
+begin
+  rw [← not_iff_not, ← int_eq_one_iff, eq_iff_le_not_lt],
+  simp only [padic_norm.of_int, true_and],
+end
+
+lemma of_nat (m : ℕ) : padic_norm p m ≤ 1 :=
+padic_norm.of_int p (m : ℤ)
+
+/-- The `p`-adic norm of a natural `m` is one iff `p` doesn't divide `m`. -/
+lemma nat_eq_one_iff (m : ℕ) : padic_norm p m = 1 ↔ ¬ p ∣ m :=
+by simp only [←int.coe_nat_dvd, ←int_eq_one_iff, int.cast_coe_nat]
+
+lemma nat_lt_one_iff (m : ℕ) : padic_norm p m < 1 ↔ p ∣ m :=
+by simp only [←int.coe_nat_dvd, ←int_lt_one_iff, int.cast_coe_nat]
+
+open_locale big_operators
+
+lemma sum_lt {t : ℚ} {n : ℕ} {F : ℕ → ℚ}
+  (hF : ∀ i, i < n → padic_norm p (F i) < t) (hn : 0 < n) :
+  padic_norm p (∑ i in finset.range n, F i) < t :=
+begin
+  induction n with d hd, {cases hn, },
+  cases d, {simp [hF], },
+  rw finset.sum_range_succ,
+  refine lt_of_le_of_lt (padic_norm.nonarchimedean p) _,
+  refine max_lt (hd (λ i hi, hF _ (hi.trans (nat.lt_succ_self _))) d.zero_lt_succ) _,
+  exact hF _ (nat.lt_succ_self _),
+end
+
+lemma sum_le {t : ℚ} {n : ℕ} {F : ℕ → ℚ}
+  (hF : ∀ i, i < n → padic_norm p (F i) ≤ t) (hn : 0 < n) :
+  padic_norm p (∑ i in finset.range n, F i) ≤ t :=
```

Comment:
I think this should be stated for sums over a nonempty `finset` in any index type, rather than for the special case of `finset.range n`. Similarly for the other such lemmas.

Reply:
Thanks! Sorry for the delay -- I've been on a family holiday.

Reply:
OK this is done.